renku/infrastructure/storage/base.py

```diff
@@ -47,12 +51,86 @@ def exists(self, uri: str) -> bool:
         else:
             return True
 
+    def mount(self, uri: str, mount_location: Path):
+        """Mount the path from the uri to a specific location locally."""
+        if not mount_location.exists():
+            raise errors.DirectoryNotFound(mount_location)
+        if not mount_location.is_dir():
+            raise errors.ExpectedDirectoryGotFile(mount_location)
+        if os.path.ismount(mount_location):
+            raise errors.ExpectedDirectoryGotMountPoint(mount_location)
+        if next(mount_location.iterdir(), None):
+            raise errors.DirectoryNotEmptyError(mount_location)
+        if not self.exists(uri):
+            raise errors.StorageObjectNotFound
+        self.set_configurations()
+        execute_rclone_command("mount", "--daemon", uri, str(mount_location.absolute()))
+
+    def get_hashes(self, uri: str, hash_type: str = "md5") -> List[FileHash]:
+        """Download hashes with rclone and parse them.
+
+        Returns a tuple containing a list of parsed hashes.
+
+        Example raw_hashes json:
+        [
+            {
+                "Path":"resources/hg19.windowmaskerSdust.bed.gz.tbi","Name":"hg19.windowmaskerSdust.bed.gz.tbi",
+                "Size":578288,"MimeType":"application/x-gzip","ModTime":"2022-02-07T18:45:52.000000000Z",
+                "IsDir":false,"Hashes":{"md5":"e93ac5364e7799bbd866628d66c7b773"},"Tier":"STANDARD"
+            }
+        ]
+        """
+        self.set_configurations()
+        hashes_raw = execute_rclone_command("lsjson", "--hash", "-R", "--files-only", uri)
+        hashes = json.loads(hashes_raw)
+        output = []
+        for hash in hashes:
+            hash_content = hash.get("Hashes", {}).get(hash_type)
+            output.append(
+                FileHash(
+                    base_uri=uri,
+                    path=hash["Path"],
+                    hash=hash_content,
+                    hash_type=hash_type if hash_content else None,
+                    modified_datetime=hash.get("ModTime"),
+                )
+            )
+        output = self._get_missing_hashes(output, hash_type=hash_type)
```

Comment:
Please see my comment above about not mounting/downloading files at this point.

Reply:
This will also be removed.